core/metadata/mongoOps.go

```diff
@@ -649,7 +648,7 @@ func mgoUpdateAddressable(ra *models.Addressable, r *models.Addressable) error {
 	if ra.Name != "" {
 		r.Name = ra.Name
 	}
-	if strings.Compare(ra.Protocol, "HTTP") != 0 || strings.Compare(ra.Protocol, "TCP") != 0 { // TODO create ENUMS that can be unmarshalled by JSON
+	if ra.Protocol != "" {
```

Comment:
Both `r` and `ra` variable names are confusing and not self-documenting.

If `ra` can take values only from a predefined set (some enum), then this comparison is not good.

Reply:
FWIW, I found from going up the call chain that "r" is the new addressable and "ra" is the old addressable. These do need to be addressed but outside of scope for this change

Reply:
That protocol property should really be an enum, but outside the scope of this change.